### PR TITLE
[OPIK-616] Add fields to LLM Provider Api Key endpoints

### PR DESCRIPTION
**Pull request by BorisTkachenko**
## Details
Add fields to LLM Provider Api Key endpoints.
Return masked raw api key.

## Testing
Added/updated corresponding integration tests

### CODE REVIEW

File: apps/opik-backend/src/test/java/com/comet/opik/podam/manufacturer/ProviderApiKeyManufacturer.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        UUID id = strategy.getTypeValue(metadata, context, UUID.class);
+
+        return ProviderApiKey.builder()
+                .id(id)
+                .provider(randomLlmProvider())
+                .apiKey(RandomStringUtils.randomAlphanumeric(20))
+                .name(strategy.getTypeValue(metadata, context, String.class))
+                .createdBy(strategy.getTypeValue(metadata, context, String.class))
+                .createdAt(Instant.now())
+                .build();
```
Comment (andrescrz) [inline apps/opik-backend/src/test/java/com/comet/opik/podam/manufacturer/ProviderApiKeyManufacturer.java:31]: I don' think you need this manufacturer. If you look at all these fields, this is standard for Podam to generate them randomly.

I believe the issue is that adding the `@Size` annotation broke the Podam generation. That's been the case for other annotations. You would just need to create a `AttributeStrategy` for `@Size.`

Anyway, not a blocker.

File: apps/opik-backend/src/test/java/com/comet/opik/podam/manufacturer/ProviderApiKeyUpdateManufacturer.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package com.comet.opik.podam.manufacturer;
+
+import com.comet.opik.api.ProviderApiKeyUpdate;
+import org.apache.commons.lang3.RandomStringUtils;
+import uk.co.jemos.podam.api.AttributeMetadata;
+import uk.co.jemos.podam.api.DataProviderStrategy;
+import uk.co.jemos.podam.common.ManufacturingContext;
+import uk.co.jemos.podam.typeManufacturers.AbstractTypeManufacturer;
+
+public class ProviderApiKeyUpdateManufacturer extends AbstractTypeManufacturer<ProviderApiKeyUpdate> {
```
Comment (andrescrz) [inline apps/opik-backend/src/test/java/com/comet/opik/podam/manufacturer/ProviderApiKeyUpdateManufacturer.java:10]: Same.

**Review by andrescrz — APPROVED** [review]
Couple of comments about the random generation for testing. But not blockers. Implementation LGTM.

File: apps/opik-backend/src/test/java/com/comet/opik/podam/manufacturer/ProviderApiKeyManufacturer.java (hunk 1 above)
Comment (BorisTkachenko) [reply apps/opik-backend/src/test/java/com/comet/opik/podam/manufacturer/ProviderApiKeyManufacturer.java:31]: Actually for `name` (with `@Size` annotation) it works fine. I added this custom manufacturer in order to increase `apiKey` length, default is 10 chars which is not enough. We mask everything less than 12 chars.

**Pull request merged**